test/hotspot/jtreg/runtime/valhalla/inlinetypes/ClassInitializationFailuresTest.java

```diff
@@ -103,45 +116,49 @@ static void testArrayInitialization() {
         // Testing anewarray when the primitive element class fails to initialize properly
         Throwable e = null;
         try {
-            BadTwo[] array = new BadTwo[10];
+            BadTwo[] array = (BadTwo[])VM.newNullRestrictedArray(BadTwo.class, 10);
         } catch(Throwable t) {
             e = t;
         }
         Asserts.assertNotNull(e, "Error should have been thrown");
         Asserts.assertTrue(e.getClass() == ExceptionInInitializerError.class, " Must be an ExceptionInInitializerError");
         // Second attempt because it doesn't fail the same way
         try {
-            BadTwo[] array = new BadTwo[10];
+            BadTwo[] array = (BadTwo[])VM.newNullRestrictedArray(BadTwo.class, 10);
         } catch(Throwable t) {
             e = t;
         }
         Asserts.assertNotNull(e, "Error should have been thrown");
         Asserts.assertTrue(e.getClass() == NoClassDefFoundError.class, "Must be a NoClassDefFoundError");
         Asserts.assertTrue(e.getCause().getClass() == ExceptionInInitializerError.class, "Must be an ExceptionInInitializerError");
+        // Transition model (annotations and array factory) doesn't permit multi-dimentional arrays tests
```

Comment:
Shouldn't support multi-dimensional arrays as well? Or are there any blockers?

Reply:
The new API to create flat arrays only supports single-dimension arrays. But because multi-dimensional arrays are basically a set of reference arrays with only the last-dimension array being an array of the element type, it is possible to create manually multi-dimensional arrays on top of this API.
Those tests were checking the behavior of `multianewarray` but with the removal of Q-descriptors, this bytecode cannot be used to create multi-dimensional arrays with a flat last-dimension array.